frame/identity/src/lib.rs

```diff
@@ -0,0 +1,1040 @@
+//!
+//! ### Dispatchable Functions
+//!
+//! #### For general users
+//! * `set_identity` - Set the associated identity of an account; a small deposit is reserved if not
+//!   already taken.
+//! * `set_subs` - Set the sub-accounts of an identity.
+//! * `clear_identity` - Remove an account's associated identity; the deposit is returned.
+//! * `request_judgement` - Request a judgement from a registrar, paying a fee.
+//! * `cancel_request` - Cancel the previous request for a judgement.
+//!
+//! #### For registrars
+//! * `set_fee` - Set the fee required to be paid for a judgement to be given by the registrar.
+//! * `set_fields` - Set the fields that a registrar cares about in their judgements.
+//! * `provide_judgement` - Provide a judgement to an identity.
+//!
+//! #### For super-users
+//! * `add_registrar` - Add a new registrar to the system.
+//! * `kill_identity` - Forcibly remove the associated identity; the deposit is lost.
+//!
+//! [`Call`]: ./enum.Call.html
+//! [`Trait`]: ./trait.Trait.html
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use rstd::prelude::*;
+use rstd::{fmt::Debug, ops::Add};
+use enumflags2::BitFlags;
+use codec::{Encode, Decode};
+use sp_runtime::{
+	traits::{StaticLookup, EnsureOrigin, Zero}, RuntimeDebug,
+};
+use support::{
+	decl_module, decl_event, decl_storage, ensure, dispatch::Result,
+	traits::{Currency, ReservableCurrency, OnUnbalanced, Get},
+	weights::SimpleDispatchInfo,
+};
+use system::{ensure_signed, ensure_root};
+
+type BalanceOf<T> = <<T as Trait>::Currency as Currency<<T as system::Trait>::AccountId>>::Balance;
+type NegativeImbalanceOf<T> = <<T as Trait>::Currency as Currency<<T as system::Trait>::AccountId>>::NegativeImbalance;
+
+pub trait Trait: system::Trait {
+	/// The overarching event type.
+	type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
+
+	/// The currency trait.
+	type Currency: ReservableCurrency<Self::AccountId>;
+
+	/// The amount held on deposit for a registered identity.
+	type BasicDeposit: Get<BalanceOf<Self>>;
+
+	/// The amount held on deposit per additional field for a registered identity.
+	type FieldDeposit: Get<BalanceOf<Self>>;
+
+	/// The amount held on deposit for a registered subaccount.
+	type SubAccountDeposit: Get<BalanceOf<Self>>;
+
+	/// The amount held on deposit for a registered subaccount.
+	type MaximumSubAccounts: Get<u32>;
+
+	/// What to do with slashed funds.
+	type Slashed: OnUnbalanced<NegativeImbalanceOf<Self>>;
+
+	/// The origin which may forcibly set or remove a name. Root can always do this.
+	type ForceOrigin: EnsureOrigin<Self::Origin>;
+
+	/// The origin which may add or remove registrars. Root can always do this.
+	type RegistrarOrigin: EnsureOrigin<Self::Origin>;
+}
+
+/// Either underlying data or a hash of it, if the data is less than 32 bytes.
+///
+/// Can also be `None`.
+#[derive(Clone, Eq, PartialEq, RuntimeDebug)]
+pub enum Data {
+	/// No data here.
+	None,
+	/// The data is stored directly.
```

Comment:
Might be good to mention that it's up to 32 bytes, and if you encode a bigger `Raw` value, the rest will be discarded.